sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/RateLimitedSampler.cs

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using OpenTelemetry.Trace;
+using System.Diagnostics;
+using System.Linq;
+
+namespace Azure.Monitor.OpenTelemetry.Exporter.Internals
+{
+    internal class RateLimitedSampler : Sampler
+    {
+        private static readonly SamplingResult RecordOnlySamplingResult = new(SamplingDecision.RecordOnly);
+        private readonly RateLimitedSamplingPercentage _samplingPercentage;
+
+        public RateLimitedSampler(double targetSpansPerSecondLimit)
+        {
+            _samplingPercentage = new RateLimitedSamplingPercentage(targetSpansPerSecondLimit);
+        }
+
+        public override SamplingResult ShouldSample(in SamplingParameters samplingParameters)
+        {
+            // 1. Respect the sampling decision of the local parent span
+
+            // get the span context of the parent span
+            var parentContext = samplingParameters.ParentContext;
+
+            SamplingResult? samplingResult = useLocalParentDecisionIfPossible(parentContext);
+            if (samplingResult != null)
+            {
+                return samplingResult.Value;
+            }
+
+            // 2. Obtain a sampling percentage (this is a number from 0 to 100)
+            double samplingPercentage = _samplingPercentage.Get();
+
+            // 3. Use the sampling percentage to make a sampling decision
+            if (samplingPercentage == 0)
+            {
+                // optimization, no need to calculate sample score in this case
+                Console.WriteLine("Sampling percentage is 0, returning RecordOnlySamplingResult.");
+                return RecordOnlySamplingResult;
+            }
+            if (samplingPercentage == 100)
+            {
+                // optimization, no need to calculate sample score in this case
```

Comment:
In .NET, we have never used the SDK sampler setting to try to avoid ingestion sampling by changing its value. Ingestion sampling is not enabled by default; customers must enable it themselves. These two concepts should not be mixed.